File: 08 CP Priority Queue/Heap Ordering/ds03_heap_order.py
import sys
class Student:
    def __init__(self, name:str, score:int) -> None:
        self.name = name
        self.score = score
    def __lt__(self, other) -> bool:
        if(self.score == other.score):
            return self.name < other.name
        return self.score > other.score
    def __str__(self) -> str:
        return self.name + " " + str(self.score) + "\n"
N,K = map(int, sys.stdin.readline().split())
students = []
for _ in range(N):
    name,score = sys.stdin.readline().strip().split()
    students.append(Student(name, int(score)))
students.sort()
for i in range(K):
    sys.stdout.write(str(students[i]))

File: 08 CP Priority Queue/Heap Ordering/ds03_heap_order.java
import java.util.*;

public class ds03_heap_order
{
    public static void main(String[] args)
    {
        Scanner Scanner = new Scanner(System.in);
        ArrayList<Student> students = new ArrayList<Student>();
        int N = Scanner.nextInt();
        int K = Scanner.nextInt();
        for(int i = 0; i < N; i++)
        {
            String name = Scanner.next();
            int score = Scanner.nextInt();
            students.add(new Student(name, score));
        }
        Scanner.close();
        Collections.sort(students);
        for(int i = 0; i < K; i++)
        {
            System.out.println(students.get(i));
        }
    }
}

class Student implements Comparable<Student>
{
    private String name;
    private int score;
    public Student(String name, int score)
    {
        this.name = name;
        this.score = score;
    }
    
    public int compareTo(Student other)
    {
        if(this.score < other.score) return 1;
        if(this.score > other.score) return -1;
        return this.name.compareTo(other.name);
    }
    
    public String toString() { return this.name + " " + this.score; }

    public String getName() { return this.name; }
    public int getScore() { return this.score; }

}